test/conformance/envpropagation_test.go

```diff
@@ -0,0 +1,161 @@
+// +build e2e
+
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/serving/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	testKey = "testKey"
+	testValue = "testValue"
+	secretName = "test-secret"
+	configMapName = "test-configmap"
+)
+
+// TestSecretsFromEnv verifies propagation of Secrets through environment variables.
+func TestSecretsFromEnv(t *testing.T) {
```

Comment:
I don't really see the scenario has any dependency on Knative, seems more on k8s side. 
@mattmoor, does the validation here align with what's intended by the original issue? just to double check.

Reply:
a further question: if it doesn't depend on knative at all (from https://kubernetes.io/docs/concepts/configuration/secret/, it seems the case indeed), should we still include it as part of Knative conformance validation? @mattmoor

Reply:
I think we specify the environment variable via the ConfigurationSpec, which exercises the controller code in Knative. But would let @mattmoor  confirm it.

Reply:
We want conformance coverage for any parts of `corev1.Container` that we allow, even if we just pass through to K8s to implement the semantics.